WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java

```diff
@@ -306,6 +307,36 @@ private void setHasFetchedMediaForAllFilters() {
         }
     }
 
+    /*
+        given a list of media, it rechecks its state against the UploadingService to see
+        if it's uploading or queued and change it accordingly, to recover from an inconsistent state
+     */
+    private List<MediaModel> getSanitizedMedia(List<MediaModel> uncuredList) {
```

Comment:
I think this is being done in the wrong place. Calling this routine every time the media grid is populated (which will happen when the activity starts, when the grid filter is changed, or when a search is performed) seems unnecessary. Wouldn't it make more sense to do this somewhere other than the media browser, since the upload states will remain incorrect until the user enters the media browser?